interfaces/builtin/avahi_observe.go

```diff
@@ -232,6 +236,8 @@ const avahiObserveConnectedPlugAppArmor = `
 # Description: allows domain, record, service, and service type browsing
 # as well as address, host and service resolving
 
+/{,var/}run/avahi-daemon/{pid,socket} rw,
```

Comment:
Why does a client of avahi need to be able to write avahi's pid file? Can you describe what the socket is used for (which might answer my question about SNAP_COMMON and SNAP_DATA)?

Reply:
indeed oversight. Client should not need to write pid file, removing it